docs/standard/parallel-programming/snippets/cs/cancellation2.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+public class CancellationTwoExample
+{
+    public static async Task Main()
+    {
+        using var cts = new CancellationTokenSource();
+        CancellationToken token = cts.Token;
+        cts.Cancel();
+
+        var task = Task.FromCanceled(token);
+        var continuation =
+            task.ContinueWith(
+                antecedent => Console.WriteLine("The continuation is running."),
+                TaskContinuationOptions.NotOnCanceled);
+
+        try
+        {
+            await task;
+        }
+        catch (Exception e)
+        {
+            static void WriteException(Exception ex) =>
+                Console.WriteLine($"{ex.GetType().Name}: {ex.Message}");
+
+            if (e is AggregateException ae)
+            {
+                foreach (Exception ie in ae.InnerExceptions)
+                {
+                    WriteException(ie);
+                }
+            }
```

Comment:
I think this can be deleted. When you `await` a Task, the `AggregateException` will be unpacked, and you catch the first exception in the `InnerExceptions` collection.